lib/active_job/cancel/queue_adapters/test_adapter.rb

```diff
@@ -0,0 +1,50 @@
+require 'active_job'
+
+module ActiveJob
+  module QueueAdapters
+    # Unfortunately we need to monkey patch the Rails TestAdapter class,
+    # because it does not save the job id on the enqueued_jobs array. We rely
+    # on a persisted id to fulfill the canceling of any given job id.
+    class TestAdapter
+      def initialize
+        activejob_version = Gem.loaded_specs['activejob'].version
```

Comment:
`ActiveJob.version` returns instance of `Gem::Version`. So if use `ActiveJob.version`, this check can more simply.

```ruby 
        if ActiveJob.version >= Gem::Version.new("5.0")
          require 'active_job/cancel/queue_adapters/test_adapter/rails_5'
        else
          require 'active_job/cancel/queue_adapters/test_adapter/rails_4'
        end
```

Reply:
Also, `require_relative` does not work correctly when symlinks are used with Ruby < 2.5.0.  
So I want to avoid to use it.
Ref: https://bugs.ruby-lang.org/issues/10222